src/EthersSafeFactory.ts

```diff
@@ -0,0 +1,118 @@
+import { Contract, Signer, Event, ethers } from 'ethers'
+import GnosisSafeProxyFactory from '@gnosis.pm/safe-contracts/build/contracts/GnosisSafeProxyFactory.json'
+import GnosisSafe from '@gnosis.pm/safe-contracts/build/contracts/GnosisSafe.json'
+import Safe from 'Safe'
+import { EMPTY_DATA, ZERO_ADDRESS } from './utils/constants'
+import EthersSafe from './EthersSafe'
+
+interface SafeProxyFactoryConfigurationSimple {
+  proxyFactoryAddress: string
+  safeSingletonAddress: string
+  data?: string
+}
+
+interface SafeProxyFactoryConfigurationExtended extends SafeProxyFactoryConfigurationSimple {
+  nonce: number
+  callbackAddress?: string
+}
+
+export type SafeProxyFactoryConfiguration =
+  | SafeProxyFactoryConfigurationSimple
+  | SafeProxyFactoryConfigurationExtended
+
+interface SafeAccountConfiguration {
+  owners: string[]
+  threshold?: number
```

Comment:
I would put this as required param (based on what you must choose in the UI outside _advanced_)